RIOT-OS/RIOT
# tools/bossa: update to latest release and adapt support

### Contribution description

This PR updates the BOSSA flashing tool to its latest release 1.9.1 which contains several enhancements:
- build on OSX is fixed so there's no need to provide a patch (removed in this PR)
- the tool now supports more SAM CPUs (SAM4S, SAMD51, et)
- no more warning during build

It also seems that the ROM_OFFSET now has to be passed explicitly when calling bossa.



### Testing procedure

This PR should be tested on boards that works with an Arduino bootloader at the beginning of the flash memory: `arduino-mkr*`, `sodaq-*`, `feather-m0` and arduino-due.

**How to test this PR:**

- remove version of bossa that could already be in `dist/tools/bossa`:
```
rm -rf ./dist/tools/bossa/{bin,bossac}
```
- put the board in bootloader mode by double tapping the reset button
- flash the board: it should succeed with this PR
```
make BOARD=arduino-mkr1000 -C tests/leds flash
```
- check that the firmware is working (red LED is blinking)

**Be careful with this PR**: it seems that omitting the ROM_OFFSET parameter can break the preflashed bootloader, since bossa is called with the `-e` (erase) option. This results in the bootloader partially erasing itself and ends up in a broken bootloader. I already bricked 2 boards because of this...
For testing, I recommend to use an Arduino MKR1000 because in case of a problem, the bootloader can be replaced in flash using JLink.
To reflash a broken bootloader on this board:
1. Download the [bootloader firmware](https://github.com/arduino/ArduinoCore-samd/blob/master/bootloaders/mkr1000/samd21_sam_ba_arduino_mkr1000.bin)
2. Plug a JLink programmer and flash the bootloader:
```
FLASHFILE=<path to>/samd21_sam_ba_arduino_mkr1000.bin make BOARD=arduino-mkr1000 -C tests/leds flash PROGRAMMER=jlink
```




### Issues/PRs references

None
